### ISSUE TITLE
[FEA] cudf::scatter_to_tables

### ISSUE THREAD

**Issue by harrism** [rapidsai/cudf#2677]
**Is your feature request related to a problem? Please describe.**

This arose from a discussion with @jrhemstad about #2452. We decided that the fastest path to a solution (time is of the essence) is to build this from existing functionality.

The requested functionality in #2452 and #2272 is effectively to group by an ID and then return the groups as a list of tables.  So it's a groupby followed by a slice.  We can all it `cudf::groups_to_tables`

**Describe the solution you'd like**

I recommend building the API:

```
std::vector<cudf::table> cudf::groups_to_tables(cudf::table const& input, cudf::column const& group_ids);
```

Internally, this will perform a groupby_without_aggregations on `input`, then a `cudf::slice` to get the rows from each group into a separate table.

Note that slice() also needs some work. It currently only operates on individual columns and returns a vector of new columns, so will need to be changed to take a table and return a vector of tables.

**Describe alternatives you've considered**
See #2452 

**Additional context**

This function is also a way to implement pandas `get_group()` although possibly overkill since Pandas does not have a function that returns a list of DataFrames, only a way to get one at a time. That said, I think it makes sense for cuDF to provide parallel versions of things Pandas does sequentially.

**Comment by karthikeyann** [comment]
> As such, I suggest the naming be `scatter_to_tables` with the API:
> `std::vector<cudf::table> cudf::scatter_to_tables(cudf::table const& input, cudf::column const& scatter_map);` 
> Row `i` in `input` will be scattered to the table indicated by `scatter_map[i]`.

I added `scatter_to_tables` in PR https://github.com/rapidsai/cudf/pull/2685

> If you'd like to preserve the original semantics that were outlined here, then this function needs to return an additional `cudf::table` that maps a table to it's `group_id`. Furthermore, `group_id` should also be extended to be a `cudf::table` to allow arbitrary rows to act as keys to identify each group.

If `scatter_to_tables` looks fine, then I can rewrite `gather_to_tables` on top of this as mentioned above. Do we really need for `gather_to_tables` now?

**Comment by karthikeyann** [comment]
@harrism @kkraus14  should I add Cython bindings as well for `groups_to_tables` and `scatter_to_tables`?

**Comment by harrism** [comment]
I think what we've been saying (perhaps not directly enough) is that `scatter_to_tables` is what we need, not `groups_to_tables`.

**Comment by karthikeyann** [comment]
`cudf::scatter_to_tables`  and Cython API `scatter_to_frames` 
PR https://github.com/rapidsai/cudf/pull/2685 merged.

**Comment by jrhemstad** [comment]
Closed by #2685
